### PR TITLE
Seffy

### PR DESCRIPTION
**Pull request by gseffy**
add report container metrics cAdvisor

### CODE REVIEW

File: org/openscore/slang/docker/maintenance/print.sl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+namespace: org.openscore.slang.docker.maintenance
```
Comment (meirwah) [inline org/openscore/slang/docker/maintenance/print.sl]: this operation can be moved to the slang/base/print folder (print is a new folder)
it makes more sense there , since this operation is not related to docker..
can you move it please?

Comment (gseffy) [inline org/openscore/slang/docker/maintenance/print.sl]: done.

**Comment by meirwah** [comment]
@levi or @orit can you please review this also?

**Comment by meirwah** [comment]
:+1:

File: org/openscore/slang/base/print/print_text.sl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+namespace: org.openscore.slang.base.print
```
Comment (oritstone) [inline org/openscore/slang/base/print/print_text.sl]: Missing copyrights section

File: org/openscore/slang/docker/maintenance/get_container_metrics_cAdvisor.sl [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      name: get_container_metrics_cAdvisor
+      inputs:
+        - host
+        - identityPort:
+            default: "'8080'"
+            required: false
+        - container
+        - url:
+            default: "'http://'+ host + ':' + identityPort +'/api/v1.2/docker/'+container"
+            override: true
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/get_container_metrics_cAdvisor.sl]: I don't think we have the *override* attribute anymore. We are using *overridable*
Did you run it with the CLI?

File: org/openscore/slang/docker/maintenance/get_machine_metrics_cAdvisor.sl [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+operation:
+      name: get_machine_metrics_cAdvisor
+      inputs:
+        - host
+        - identityPort:
+            default: "'8080'"
+            required: false
+        - url:
+            default: "'http://'+ host + ':' + identityPort +'/api/v1.2/machine'"
+            override: true
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/get_machine_metrics_cAdvisor.sl]: Same comment about the overiide attribute

File: org/openscore/slang/docker/maintenance/get_machine_metrics_cAdvisor.sl [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+####################################################
+#   This operation will do a REST call to cAdvisor runs in a docker its response
+#   (result needs to be parsed).
+#
+#   Inputs:
+#       - host - docker machine host
+#       - identityPort - optional - port used for cAdvisor - Default: 8080
+#       - container - name or ID of the Docker container that runs cAdvisor
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/get_machine_metrics_cAdvisor.sl]: I don't see that input in the operation...

File: org/openscore/slang/docker/maintenance/parse_cadvisor_container.sl [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+#
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+####################################################
+#   This operation will parse the response of the cAdviser container information and have the
+#   decoded as output.
+#
+#   Inputs:
+#       -jsonResponse - response of the cAdviser container information
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/parse_cadvisor_container.sl]: Missing documenattion for the *machine_memory_limit* input

File: org/openscore/slang/docker/maintenance/parse_cadvisor_container.sl [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+#       - SUCCESS - parsing was successful (returnCode == '0')
+#       - FAILURE - otherwise
+####################################################
+
+namespace: org.openscore.slang.docker.maintenance
+
+operation:
+      name: parse_cadvisor_container
+      inputs:
+        - jsonResponse
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/parse_cadvisor_container.sl]: We use a convension of loswer case words separated by _ for inputs, outputs,,, wherever possible

File: org/openscore/slang/docker/maintenance/parse_cadvisor_container.sl [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+#   This operation will parse the response of the cAdviser container information and have the
+#   decoded as output.
+#
+#   Inputs:
+#       -jsonResponse - response of the cAdviser container information
+#   Outputs:
+#       - decoded - parse response
+#       - returnResult - notification string which says if parsing was successful or not
+#       - returnCode - 0 if parsing was successful, -1 otherwise
+#       - errorMessage - returnResult if there was an error
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/parse_cadvisor_container.sl]: Missing many outputs from the documentation

File: org/openscore/slang/docker/maintenance/parse_cadvisor_machine.sl [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+#       - SUCCESS - parsing was successful (returnCode == '0')
+#       - FAILURE - otherwise
+####################################################
+
+namespace: org.openscore.slang.docker.maintenance
+
+operation:
+      name: parse_cadvisor_machine
+      inputs:
+        - jsonResponse
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/parse_cadvisor_machine.sl]: Use lower case words for the input name

File: org/openscore/slang/docker/maintenance/parse_cadvisor_machine.sl [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+#   This operation will parse the response of the cAdviser container information and have the
+#   decoded as output.
+#
+#   Inputs:
+#       -jsonResponse - response of the cAdviser container information
+#   Outputs:
+#       - decoded - parse response
+#       - returnResult - notification string which says if parsing was successful or not
+#       - returnCode - 0 if parsing was successful, -1 otherwise
+#       - errorMessage - returnResult if there was an error
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/parse_cadvisor_machine.sl]: Missing outputs

File: org/openscore/slang/docker/maintenance/report_container_metrics_cAdvisor.sl [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+#
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+##################################################################################################################################################
+#   This flow retrieves cAdviser status of a container in docker 
+#
+#   Inputs:
+#       - container - name or ID of the Docker container that runs MySQL
+#       - dockerHost - Docker machine host
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/report_container_metrics_cAdvisor.sl]: - Wrong input name
- lower case words with _

File: org/openscore/slang/docker/maintenance/report_machine_metrics_cAdvisor.sl [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+#   The Apache License is available at
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+##################################################################################################################################################
+#   This flow retrieves cAdviser status of a container in docker 
+#
+#   Inputs:
+#       - container - name or ID of the Docker container that runs MySQL
+#       - dockerHost - Docker machine host
+#       - identityPort - optional - port used for cAdvisor - Default: 8080
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/report_machine_metrics_cAdvisor.sl]: - Wrong input name
- Missing inputs
- lower case words with _
- Missing outputs & results from doc

**Comment by oritstone** [comment]
@meirwah , did you review the functionality of the content? I only went over the documentation, and content structure guidelines

**Comment by meirwah** [comment]
yes...

**Comment by meirwah** [comment]
:+1:

**Comment by meirwah** [comment]
just 1 more thing, can you move all cAdvisor related flows to a cAdvisor dir, instead of maintence , i think it is more correct...

File: org/openscore/slang/docker/maintenance/parse_cadvisor_machine.sl [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+        - json_response
+      action:
+        python_script: |
+          try:
+            import json
+            decoded = json.loads(json_response)
+            num_cores=int(decoded['num_cores'])
+            cpu_frequency_khz=int(decoded['cpu_frequency_khz'])
+            memory_capacity=int(decoded['memory_capacity'])
+            filesystems=decoded['filesystems']
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/parse_cadvisor_machine.sl]: seems to me that you have a typeo in filesystems. You refer to it as file_systems in the output.
Did you test it is actually running correctly?

File: org/openscore/slang/docker/maintenance/parse_cadvisor_machine.sl [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+#       - num_cores - machine number of cores
+#       - cpu_frequency_khz - machine cpu
+#       - memory_capacity- machine memory
+#       - file_systems - parse cAdviser machine filesystems
+#       - disk_map- parse cAdviser machine disk map
+#       - network_devices- parse cAdviser machine network devices
+#       - topology- parse cAdviser machine topology
+#       - returnResult - notification string which says if parsing was successful or not
+#       - returnCode - 0 if parsing was successful, -1 otherwise
+#       - errorMessage - returnResult if there was an error
```
Comment (oritstone) [inline org/openscore/slang/docker/maintenance/parse_cadvisor_machine.sl]: Those outputs should be with lower case letters

**Comment by oritstone** [comment]
For print_text.sl - missing copyrights and doc
